README.md

```diff
@@ -1,12 +1,114 @@
+(Build the release version to save **a lot** of RAM).
+
+`llvm-tblgen` should be in `<somewhere>/llvm-project/bin/` after the build.
+
+### Build and move Hexagon.json
+
+```bash
+cd llvm-project/llvm/lib/Target/Hexagon
+llvm-tblgen -I ../../../include/ --dump-json -o Hexagon.json Hexagon.td
+mv Hexagon.json <path to>/rz-hexagon/
+```
+
+# Install
+
+```bash
+cd rz-hexagon/
+pip install -r requirements.txt
+# If you enjoy some colors
+pip install -r optional_requirements.txt
+# Run tests
+cd Tests
+python3 -m unittest discover -s . -t .
+# Install as develop package
+cd ..
+pip install -e .
+```
+
+# Generate PlugIn
+
+Simply run:
+```
+./LLVMImporter.py
+```
+
+It processes the files and generates C code in `./rizin` and its subdirectories.
+
+Copy the generated files to the `rizin` directory with
+  ```commandline
+  rsync -a rizin/ <rz-src-path>/
+  ```
+
+# Porting
+
+Apart from some methods, which produce the C code for `rizin`, this code is `rizin` independent.
+In theory, it shouldn't be that hard to use it for disassembler plugins of other reverse engineering frameworks.
+
+So here are some good to know points for porting:
+- All `rizin` specific methods have the leading comment: `# RIZIN SPECIFIC`.
+- Please open an issue if you start working on this code for another reverse engineering framework.
+  We could remove all `rizin` code from this repo and fork our framework specific plugins from it.
+- Specifics about the framework code (line indent, variable names etc.) can be set in `PluginInfo.py`
+
+# Development info
+
+- The best way to start is to take a look at an instruction in `Hexagon.json`.
+  We take all information from there and knowing the different objects
+  makes it easier to understand the code.
+- If you need any information about a llvm specific term or variable name from the `Hexagon.json` file a simple
+  `grep -rn "term" llvm-project/llvm/lib/Target/Hexagon/` will usually help.
+- If you parse LLVM data always end it with an exception else statement:
+  ```python
+  if x:
+     ...
+  elif y:
+     ...
+  elif z:
+     ...
+  else:
+    raise ImplementationException("This case seems to be new, please add it.")
+  ```
+- Names of variables which holds data directly taken from the `Hexagon.json` file
+  should have a name which starts with
+`llvm_`.
+  
+  For example:
+  
+  - `llvm_in_operands` holds a list with the content of `Hexagon.json::[Instr].InOperandList`.
+  - `llvm_syntax` holds: `$Rdd8 = combine(#0,#$Ii) ; $Rx16 = add($Rx16in,$Rs16)` (the syntax in LLVM style).
+
+    In case of this duplex Instruction it is actually the LLVM-syntax of the high and
+    low instruction concatenated with a semicolon.
+  - `syntax` holds: `Rdd = combine(#0,#Ii) ; Rx = add(Rxin,Rs)`
+  - `Instruction.operands` is a dictionary which contains `Register` and `Immediate` Python objects.
 
-The parsing part based largely on [hexag00n](https://github.com/programa-stic/hexag00n) importer.
 
 # Contributors
 
```

Comment:
Maybe add yourself here ;)

Reply:
And obviously at the first place since most of the code is yours now.

Reply:
In fact, remove everyone else, except yourself, since you rewrote everything from scratch anyway.